### ISSUE TITLE
makejdk-any-platform.sh -s, -d parameters do not work as they are supposed to

### ISSUE THREAD

**Issue by sxa** [AdoptOpenJDK/openjdk-build#2058]
Using `-s` which is expected to be given as an absolute PATH does not work - it gets appended to the workspace directory where `makejdk-any-platform.sh` is invoked. This example output was from trying to set them both to directories under `/home/sxa/jdk11`:
```
Didn't find any existing openjdk repository at /home/sxa/openjdk-build/workspace/home/sxa/jdk11/src//home/sxa/jdk11/src so cloning the source to openjdk
git clone --depth=1 https://github.com/adoptopenjdk/openjdk-jdk11u.git /home/sxa/openjdk-build/workspace//home/sxa/jdk11/src/src
Cloning into '/home/sxa/openjdk-build/workspace//home/sxa/jdk11/src/src'...
checkoutRequiredCodeToBuild:
  workspace = /home/sxa/openjdk-build/workspace//home/sxa/jdk11/src/src
  BUILD_VARIANT = hotspot
  TAG = 
  BRANCH = dev
```
Same is true of `-d` rendering both more or less useless.

**Comment by sxa** [comment]
Also `-T` cannot be a full path to the location to put the resulting tarball - if you try you get a message like this:
```
Moving the artifact to /home/sxa/openjdk-build/workspace//home/sxa/jdk11/build
mv: cannot move `/home/sxa/openjdk-build/workspace/home/sxa/jdk11/src/src/build/linux-x86_64-normal-server-release/images/OpenJDK.tar.gz' to `/home/sxa/openjdk-build/workspace//home/sxa/jdk11/build//home/sxa/MyOpenJDK11.tar.gz': No such file or directory
```

**Comment by hendrikebbers** [comment]
@sxa Can you add some information how the error can be reproduces? If the error can be easily reproduced by a newcomer we can create a "good first issue" out of this one :)

**Comment by andrew-m-leonard** [comment]
BUILD_ARGS of
--destintion
--source
don't act as expected.
We need to check the behavior of using the makejdk-any-platform.sh with these options:
see: https://github.com/adoptium/temurin-build/blob/8e043e364ecac19f72fc592116de9386ed7ffd0d/sbin/common/config_init.sh#L263

**Comment by andrew-m-leonard** [comment]
Needs investigation as to whether these arguments work?

**Comment by julian55455** [comment]
Hi @smlambert can I take on this?

**Comment by julian55455** [comment]
> Needs investigation as to whether these arguments work?

@andrew-m-leonard which command do we have to run with these parameters?

**Comment by zdtsw** [comment]
> > Needs investigation as to whether these arguments work?
> 
> @andrew-m-leonard @zdtsw which command do we have to run with these parameters?

`makejdk-any-platform.sh `  `-s` and `-d ` are the parameters for it.

**Comment by julian55455** [comment]
> `makejdk-any-platform.sh ` `-s` and `-d ` are the parameters for it.

@zdtsw Wondering :open_mouth:  why it isn't picking when I run it from the root directory of the project

**Comment by zdtsw** [comment]
> > `makejdk-any-platform.sh ` `-s` and `-d ` are the parameters for it.
> 
> @zdtsw Wondering 😮 why it isn't picking when I run it from the root directory of the project

Care to post your log? how you run it?

**Comment by julian55455** [comment]
> Care to post your log? how you run it?

![git](https://user-images.githubusercontent.com/78744671/195400428-9f23d884-aac3-4265-94b8-1246e94e7277.png)


@zdtsw there it is

**Comment by zdtsw** [comment]
> > Care to post your log? how you run it?
> 
> ![git](https://user-images.githubusercontent.com/78744671/195400428-9f23d884-aac3-4265-94b8-1246e94e7277.png)
> 
> @zdtsw there it is

permisson on the file? change to executable and see how it goes without `./`

**Comment by julian55455** [comment]
> permisson on the file? change to executable and see how it goes without

@zdtsw  Actually I had checked earlier and it has all the file rights read, write and execute

**Comment by julian55455** [comment]
> permisson on the file? change to executable and see how it goes without

@zdtsw  Actually I had checked earlier and it has all the file rights read, write and execute

**Comment by zdtsw** [comment]
> > permisson on the file? change to executable and see how it goes without
> 
> @zdtsw Actually I had checked earlier and it has all the file rights read, write and execute
> 
> Btw am running Ubuntu 20.04 ![git2](https://user-images.githubusercontent.com/78744671/195510828-2543e829-94c7-4831-8ba2-072ea1edd370.png)
> 
> @sxa @zdtsw I actaully tried to spin from another machine today and it didn't help. Could this be another issue?

I do not think this is an issue. At least to call a script, I always use the path or add `sh` (rarely)
The only case as far as I know is, the executable has been added into the PATH, or it is a linux built-in command, which can skip `./`

**Comment by julian55455** [comment]
@sxa can help us reproduce this bug possibly

**Comment by sxa** [comment]
makejdk-any-platform.sh requires a parameter to specify which JDK to build. If you run it without any parameters it will say:

```
[sxa@fedora openjdk-build]$ ./makejdk-any-platform.sh
Starting ./makejdk-any-platform.sh to configure, build (Adoptium)OpenJDK binary
Please provide a java version to build as an argument
[sxa@fedora openjdk-build]$ 
```
So the normal thing to do would be to run `./makejdk-any-platform.sh jdk`.

To use `-s` (which is to specify a directory on your local machine to clone to) you need to tell it which directory to use for the `-s` option e.g. 
```
[sxa@fedora openjdk-build]$ ./makejdk-any-platform.sh
Starting ./makejdk-any-platform.sh to configure, build (Adoptium)OpenJDK binary
Please provide a java version to build as an argument
[sxa@fedora openjdk-build]$ 
```
If you run `./makejdk-any-platform.sh -h` it will give you more information on the parameters, or just look at the source of that help page which is in https://github.com/adoptium/temurin-build/blob/master/makejdk-any-platform.1

**Comment by andrlos** [comment]
https://github.com/adoptium/temurin-build/pull/4514 @smlambert or @sxa or anyone .. can you take a look?
